v1/config/sites.ts

```diff
@@ -1,41 +1,105 @@
-import { NewSite, UpdateSite, Site } from '../api/schemas'
+import { NewSite, ProtocolStatus, Site } from '../api/schemas.js'
 import { Static } from '@sinclair/typebox'
 import { Config } from './store.js'
-import { nanoid } from 'nanoid'
-
-export const DEFAULT_SITE_CFG = {
-  dns: { // TODO: what is a good default DNS to use here?
-    server: '',
-    domains: []
-  },
-  links: {}
-}
+import { AbstractLevel } from 'abstract-level'
+import { ProtocolManager } from '../protocols/index.js'
+import { Ctx } from '../protocols/interfaces.js'
+import isValidHostname from 'is-valid-hostname'
 
 export class SiteConfigStore extends Config<Static<typeof Site>> {
+  protocols?: ProtocolManager
+
+  constructor (db: AbstractLevel<any, string, any>, protocols?: ProtocolManager) {
+    super(db)
+    this.protocols = protocols
+  }
+
   async create (cfg: Static<typeof NewSite>): Promise<Static<typeof Site>> {
-    const id = nanoid()
-    const obj = {
+    const id = cfg.domain
+    if (!isValidHostname(id)) {
+      return await Promise.reject(new Error('Invalid hostname. Please ensure you leave out the port and protocol specifiers (e.g. no https://)'))
```

Comment:
Does `throw` not work for this case? If it's a linter thing, I'd appreciate it if we could disable whatever rule is mandating this code style. :P